app/src/main/java/org/simple/clinic/settings/SettingsEffectHandler.kt

```diff
@@ -65,4 +68,16 @@ class SettingsEffectHandler @AssistedInject constructor(
           }
     }
   }
+
+  private fun checkAppUpdateAvailability(
+      appUpdateAvailability: CheckAppUpdateAvailability
+  ): ObservableTransformer<CheckAppUpdateAvailabilityEffect, SettingsEvent> {
+    return ObservableTransformer { effectStream ->
+      effectStream
+          .flatMap {
```

Comment:
This should be a `.switchMap` instead since we would want to unsubscribe from the previous subscription if the effect is received.